mdaviz/mainwindow.py

```diff
@@ -76,116 +83,157 @@ def doClose(self, *args, **kw):
-        self.files.clear()
-        self.files.addItems(files_list)   
-        
-    def setFile(self,mda_file):
-        print(f"{mda_file=}")
-        full_path_str=self.folderName()+'/'+mda_file
-        self.setStatus(f"Selected file {full_path_str!r}.")      
-        self._mdaFileName=mda_file
-    
-    def setFolderPath(self, folder_name = DATA_FOLDER):
-        """A folder was selected (from the open dialog)."""
-        # TODO: check for validity (does it has mda in there?)
-        # if len(catalog_name) == 0 or catalog_name not in self.server():
-        #     if len(catalog_name) > 0:
-        #         self.setStatus(f"Catalog {catalog_name!r} is not supported now.")
-        #     return
-
-        folder_path = Path(folder_name)
-
-        self._folderPath = folder_path
-        self._folderName = folder_name
+    def setmdaFileList(self,data_path):
+        if data_path:
+            self._mdaFileList = sorted([file.name for file in data_path.glob('*.mda')])
+        else:
+            self._mdaFileList = None
+
+    def setSubFolderName(self):
+        self._subFolderPath = Path(self.subfolder.currentText())
+            
+    def setSubfolderList(self, subfolder_list):
+        """Set the subfolders path list in the pop-up list."""
+        self._subFolderList = subfolder_list
+        self.subfolder.clear()
+        self.subfolder.addItems(subfolder_list)
+
+    def setFolderPath(self, folder_name):
+        """A folder was selected (from the open dialog)."""   
         
-        # FIXME the is bellow is not longer very relevant since we now convert the str to path just above; need to check for validity above
-        if isinstance(folder_path,Path):   
-            from .mda_folder import MDA_MVC
-
-            # TODO: check if the folder has mda files in it 
-            spec_name = MDA_SPEC_NAME
-            self.spec_name.setText(spec_name)
-            # TODO do we really want to display that or is there a more relevant message to display? (eg valid folder?)
-            self.setStatus(f"Folder path: {folder_name!r}")
+        if folder_name == "Other...":
+            self.doOpen() 
+                    
+        else:
             
-            self.setmdaFileList(folder_path)
-            mda_list = self.mdaFileList()
-            self.setFiles(mda_list)
+            folder_path = Path(folder_name)
+            layout = self.groupbox.layout()    
 
-            layout = self.groupbox.layout()
-            self.clearContent(clear_file=False)
-                
-            self.mvc_folder = MDA_MVC(self)
-            layout.addWidget(self.mvc_folder)
+            if folder_path.exists() and folder_path.is_dir():   # folder exists
+                self._folderPath = folder_path
 
-        else:
-            self.mvc_folder = None
-            layout.addWidget(QtWidgets.QWidget())  # nothing to show
+                def get_all_subfolders(folder_path, parent_path=""):
```

Comment:
This does not have the exact behavior I am looking for. It yields this:
<img width="1282" alt="image" src="https://github.com/BCDA-APS/mdaviz/assets/39918069/bd307266-1ddf-4c16-928d-ddd52f72dee1">

Instead of this:
<img width="1286" alt="image" src="https://github.com/BCDA-APS/mdaviz/assets/39918069/94fe26d6-3340-408f-bc70-b5abf9bb56ce">


To make the list comprehension do what I want would take a bit of thinking I would rather look into this in a separate issue #27